spacy/tokens/doc.pyx

```diff
@@ -1451,6 +1451,140 @@ cdef class Doc:
                 remove_label_if_necessary(attributes[i])
                 retokenizer.merge(span, attributes[i])
 
+    def from_json(self, doc_json):
+        """Convert a JSON document generated by Doc.to_json() to a Doc.
+
+        vocab (Vocab): A storage container for lexical types.
+        RETURNS (Doc): A doc instance corresponding to the specified JSON representation.
+        """
+
+        ### Token-level properties ###
+
+        words = []
+        annotation_types = ("pos", "head", "dep", "lemma", "tag", "morph")
```

Comment:
Agreed. As far as I can tell, there isn't a good reason to use the lowercase strings. I suggest using the variables everywhere in `from_json()`.

Reply:
Suggestion for using `[POS, ...]` instead of string representations in latest commit. There is still an initial mapping to the string representations, but we won't be able to bypass that if we don't wanna change the output of `Doc.to_json()`.